### PR TITLE
Add push mobile plugin

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: wazo_webhookd/services/mobile/plugin.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+    @classmethod
+    def get_external_token(cls, config, user_uuid):
+        auth = cls.get_auth(config)
+        token = auth.external.get('mobile', user_uuid)
+        tenant_uuid = auth.users.get(user_uuid)['tenant_uuid']
+        external_config = auth.external.get_config('mobile', tenant_uuid)
+
+        if token["apns_token"]:
+            external_config['ios_apns_cert'] = '/tmp/ios.pem'
```
Comment (pc-m) [inline wazo_webhookd/services/mobile/plugin.py]: All tenants use the same file `/tmp/ios.pem`

which could mix the values for different tenants when using the value stored in the file

File: wazo_webhookd/services/mobile/plugin.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+
+class PushNotification(object):
+
+    def __init__(self, external_token, apns_token, external_config):
+        self.token = external_token
+        self.apns_token = apns_token
+        self.external_config = external_config
+
+    def send_notification(self, data):
```
Comment (pc-m) [inline wazo_webhookd/services/mobile/plugin.py]: I'd rather have 3 methods
`send_incoming_call_notification`, `send_voicemail_received_notification`, `send_chat_msg_notification` and avoid the if here

File: wazo_webhookd/services/mobile/plugin.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+                    extra_notification_kwargs=dict(android_channel_id=channel_id),
+                    data_message=data)
+
+            if notification.get('failure') != 0:
+                logger.error('Error to send push notification: %s', notification)
+            return notification
+
+    def _send_via_apn(self, apns_token, data):
+        payload = Payload(alert=data, sound="default", badge=1)
+        client = APNsClient(self.external_config['ios_apns_cert'],
```
Comment (pc-m) [inline wazo_webhookd/services/mobile/plugin.py]: I don't know what could happen here but the content of `/tmp/ios.pem` is the last one that was written to the file. Which could be for another tenant.

**Comment by sileht** [comment]
Since it's only code move here I open https://wazo-dev.atlassian.net/browse/WAZO-922 to fix your comment

**Pull request merged**